### PR TITLE
consolidate existing CIs

### PR DESCRIPTION
**Pull request by mingwli**
seems yaml syntax is different when it's not in `condition` expression

🔁 🚗 💥

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -12,21 +12,26 @@ exec "dotnet test test\docfx.Test"
...
+if ($branch -eq "refs/heads/v3") {
+    # CI triggered by v3
     $version = "3.0.0-beta-$revision-$commitSha"
 } elseif ($branch.StartsWith($featureBranchPrefix)) {
+    # CI triggered by feature/*
     $feature = $branch.SubString($featureBranchPrefix.length)
     $version = "3.0.0-alpha-$feature-$revision-$commitSha"
 } else {
+    # local run
+    $branch = $branch.replace('/', '-')
```
Comment (928PJY) [inline build.ps1]: never used?

File: .deploy.yml [hunk 2]
```diff
@@ -1,34 +1,58 @@
...
+- feature/*
+jobs:
+# unit & e2e test jobs
+- job: Linux Test(ubuntu-16.04)
+  pool:
+    vmImage: 'ubuntu-16.04'
+  condition: eq(variables['Build.Reason'], 'PullRequest')
+  steps:
+    - bash: ./build.sh
+- job: macOS Test(macOS-10.13)
```
Comment (OsmondJiang) [inline .deploy.yml]: any of the build job fails the whole build fails?

File: build.ps1 [hunk 3]
```diff
@@ -12,22 +12,27 @@ exec "dotnet test test\docfx.Test"
...
-    $feature = $branch.SubString($featureBranchPrefix.length)
-    $version = "3.0.0-alpha-$feature-$revision-$commitSha"
+} elseif ($branch.StartsWith($branchPrefix)) {
+    # CI triggered by any other branch
+    $branch = $branch.SubString($branchPrefix.length).Replace('/', '-')
+    $version = "3.0.0-alpha-$branch-$revision-$commitSha"
 } else {
-    $version = "3.0.0-alpha-$revision-$commitSha"
+    # local run
+    $branch = $branch.Replace('/', '-')
```
Comment (OsmondJiang) [inline build.ps1]: will there be too many packages versions? is there any limitation of MyGet?

File: .deploy.yml (hunk 2 above)
Comment (mingwli) [reply .deploy.yml]: I haven't found the doc, but during my testing it happened once.

**Pull request merged**